catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -448,6 +455,22 @@ def get_value_by_names(key_value_list: list, prop_names: list):
             ):
                 return val
 
+    @staticmethod
+    def get_filetype(url, media_type) -> str | None:
+        """
+        Extract the filetype from extension in the media url.
+
+        Returns only if the media type guessed matches the expected media type.
+        """
+        filetype, extracted_media_type = extract_filetype(url)
+        if extracted_media_type != media_type:
```

Comment:
Is this something that can be generically applied to all providers?

Reply:
That is my intention in including it in the `MediaStore` class (which you also commented on). The difference is that there, it should raise an Exception (so we're aware of the problem), and here on Wikimedia, it should simply skip the miscategorized record. That's my view, at least. Do you picture another approach to this?

Reply:
> here on Wikimedia, it should simply skip the miscategorized record

Using a ValueError in this comment to communicate that makes it difficult to understand that that is the intention; just follows into the other comment I shared about how this control flow is managed compared to the other checks.